### PR TITLE
[DataGrid] Fix strict mode issue with apiRef

### LINKED ISSUE #849: [XGrid] ViewPort doesn't update on table state change when using the useApiRef in React.StrictMode
- [x ] The issue is present in the latest release.
- [x ] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
When using the useApiRef in React strictMode - the XGrid stops re-rendering view - but table state updates.
 The re-render can then be triggered by resizing the window. 

 
 

## Expected Behavior 🤔

Grid should keep current functionality and provide extra flexibility through the api. 

## Steps to Reproduce 🕹

 

 

https://codesandbox.io/s/sad-tu-tn5zs

Steps:

 1. Click e.g sort. 
 2. Nothing happens.
 3. Resize the window.
 4. Sorting applies

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
  Don't forget to mention which browser you used.
  Output from `npx @material-ui/envinfo` goes here.
```
</details>

### PR DESCRIPTION
**Pull request by dtassone**
I have activated strict mode on these 2 stories. The main issue occurred when passing the apiRef as a prop. It is now fixed
http://localhost:6006/?path=/story/x-grid-tests-state--strict-demo
http://localhost:6006/?path=/story/x-grid-tests-pagination--pagination-api-tests

Todo: 
- [x] Activate Strict mode on all Unit tests
- [x] Activate Strict mode for all storybook

Fix #849 
Fix codesandbox https://codesandbox.io/s/create-react-app-forked-kcoup?file=/src/App.js

### CODE REVIEW

**Review by oliviertassinari — CHANGES_REQUESTED** [review]
Please update the test cases to remove `strict: false`/write a new test case if the linked issue is not covered. It's important we test strict mode in the CI.

**Review by DanailH — APPROVED** [review]

**Review by oliviertassinari — APPROVED** [review]
Great

File: packages/storybook/.storybook/preview.tsx [hunk 1]
```diff
@@ -38,3 +37,11 @@ export const parameters = {
     page: null,
   },
 };
+
+export const decorators = [
+  (Story) => (
+    <React.StrictMode>
```
Comment (oliviertassinari) [inline packages/storybook/.storybook/preview.tsx:43]: So not using https://github.com/storybookjs/storybook/issues/12734 😆

Comment (dtassone) [reply packages/storybook/.storybook/preview.tsx:43]: Nope! I tried but it didn't seem to work

**Pull request merged**